modules/rostests/apitests/win32nt/ntuser/NtUserSetTimer.c

```diff
@@ -1,18 +1,247 @@
+    {
+        UINT_PTR locIndex = SetTimer(NULL, 0, TEST2_INTERVAL, TimerProc);
+
+        if (locIndex == 0)
+            countErrors++;
+        if (KillTimer(NULL, locIndex) == 0)
+            countErrors++;
+    }
+
+    return (countErrors == 0);
+}
+
+// TEST WITH MESSAGES WITHOUT WINDOW - test different ids
+static BOOL test3(void)
+{
+    UINT countErrors = 0;
+
+    UINT_PTR locIndex1 = SetTimer(NULL, 0, TEST1_INTERVAL, TimerProc);
+    if (locIndex1 == 0)
+        countErrors++;
+    if (KillTimer(NULL, locIndex1) == 0)
+        countErrors++;
+    UINT_PTR locIndex2 = SetTimer(NULL, 0, TEST1_INTERVAL, TimerProc);
+    if (locIndex2 == 0)
+        countErrors++;
+    if (KillTimer(NULL, locIndex2) == 0)
+        countErrors++;
+    if (locIndex1 == locIndex2)
+        countErrors++;
+
+    return (countErrors == 0);
+}
+
+// TEST WITH MESSAGES WITH WINDOW - test count of sent messages
+static BOOL testW1(HWND hwnd)
+{
+    UINT i, countErrors = 0;
+
+    if (hwnd == NULL)
+        return FALSE;
+
+    int minMessages = ((float)SLEEP_TIME / (float)TESTW1_INTERVAL) * (1 - TIME_TOLERANCE);
+    int maxMessages = ((float)SLEEP_TIME / (float)TESTW1_INTERVAL) * (1 + TIME_TOLERANCE);
+
+    ZeroMemory(timerIdW1, sizeof(timerIdW1));
+
+    for (i = 0; i < TESTW1_COUNT; i++)
+    {
+        UINT_PTR locIndex = SetTimer(hwnd, i, TESTW1_INTERVAL, NULL);
+        if (locIndex == 0)
+            countErrors++;
+    }
+
+    ULONGLONG startTime = GetTickCount();
+
+    MSG msg = { NULL };
+    while (GetMessage(&msg, NULL, 0, 0))
+    {
+        TranslateMessage(&msg);
+        DispatchMessage(&msg);
+
+        if (GetTickCount() - startTime >= SLEEP_TIME)
+            PostQuitMessage(0);
+    }
+
+    for (i = 0; i < TESTW1_COUNT; i++)
+    {
+        if ((timerIdW1[i].counter < minMessages) || (timerIdW1[i].counter > maxMessages))
+            countErrors++;
+    }
+
+    for (i = 0; i < TESTW1_COUNT; i++)
+    {
+        if (KillTimer(hwnd, i) == 0)
+            countErrors++;
+    }
+
+    return (countErrors == 0);
+}
+
+// TEST WITH MESSAGES WITH WINDOW - create many timers
+static BOOL testW2(HWND hwnd)
+{
+    UINT i, countErrors = 0;
+
+    if (hwnd == NULL)
+        return FALSE;
+
+    for (i = 0; i < TESTW2_COUNT; i++)
+    {
+        UINT_PTR result = SetTimer(hwnd, 1, TESTW2_INTERVAL, NULL);
+        if (result == 0)
+            countErrors++;
+        if (KillTimer(hwnd, 1) == 0)
+            countErrors++;
+    }
+
+    return (countErrors == 0);
+}
+
 START_TEST(NtUserSetTimer)
 {
+    // TEST WITH MESSAGES WITHOUT WINDOW - test count of sent messages
+    TEST(test1());
+
+    // TEST WITH MESSAGES WITHOUT WINDOW - create many timers
+    TEST(test2());
+
+    // TEST WITH MESSAGES WITHOUT WINDOW - test different ids
+    TEST(test3());
+
+    WNDCLASS wc = { 0 };
+    wc.lpfnWndProc = WindowProc;
+    wc.hInstance = GetModuleHandle(NULL);
+    wc.lpszClassName = "TimerWindowClass";
+    RegisterClass(&wc);
```

Comment:
```suggestion
    wc.lpszClassName = L"TimerWindowClass";
    RegisterClassW(&wc);
```
I wonder whether we should check the returned value from RegisterClass ...